dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/mail/MailMessageCreator.java

```diff
@@ -403,53 +475,89 @@ private String authenticatedUserMessage(EmailNotificationDto notificationDto, St
         return StringUtils.EMPTY;
     }
 
-    private String getSubjectText(NotificationType notificationType, String rootDataverseName) {
+    private String getSubjectText(EmailNotificationDto notificationDto, String rootDataverseName) {
         List<String> rootDvNameAsList = Collections.singletonList(rootDataverseName);
-        switch (notificationType) {
+        switch (notificationDto.getNotificationType()) {
             case ASSIGNROLE:
-                return BundleUtil.getStringFromBundle("notification.email.assign.role.subject", rootDvNameAsList);
+                return BundleUtil.getStringFromBundle("notification.email.assign.role.subject",
+                                                      notificationDto.getNotificationReceiver().getNotificationsLanguage(),
```

Comment:
Maybe we can put this `notificationDto.getNotificationReceiver().getNotificationsLanguage()` into some variable, and reduce repetitions in following lines?